src/elements/content-sidebar/activity-feed/app-activity/AppActivity.js

```diff
@@ -68,54 +65,107 @@ function mapActivityNodes(node: HTMLLinkElement): React.Node {
+        const doc: Document = this.parser.parseFromString(renderedText, 'text/html');
+        if (!doc) {
+            return [];
+        }
+
+        const childNodes = getProp(doc, 'body.childNodes', []);
+        return Array.from(childNodes);
+    };
+
+    render() {
+        const {
+            activity_template: { id: templateId },
+            app: { name, icon_url },
+            created_at: createdAt,
+            created_by: createdBy,
+            currentUser,
+            error,
+            intl,
+            isPending,
+            permissions,
+        } = this.props;
+
+        const canDelete = getProp(permissions, 'can_delete', false) || currentUser.id === createdBy.id;
+        const createdAtTimestamp = new Date(createdAt).getTime();
+        const isMenuVisible = canDelete && !isPending;
+        const { isConfirmingDelete } = this.state;
+
+        return (
+            <Media
+                className={classNames('bcs-AppActivity', {
                     'bcs-is-pending': isPending || error,
                 })}
+                data-resin-target="loaded"
+                data-resin-feature={`app_activity_${templateId}_card`}
             >
-                <img
-                    className="bcs-app-activity-icon"
-                    alt={intl.formatMessage(messages.appActivityAltIcon, { appActivityName: name })}
-                    src={icon_url}
-                />
-                <div className="bcs-app-activity-content">
-                    <div className="bcs-app-activity-headline">
-                        <span className="bcs-app-activity-app-name">{name}</span>
-                        {canDelete && !isPending && (
-                            <InlineDelete
-                                id={id}
-                                permissions={permissions}
-                                message={<FormattedMessage {...messages.appActivityDeletePrompt} />}
-                                onDelete={onDelete}
-                            />
-                        )}
-                    </div>
+                <Media.Figure>
+                    <img
+                        className="bcs-AppActivity-icon"
+                        alt={intl.formatMessage(messages.appActivityAltIcon, { appActivityName: name })}
+                        src={icon_url}
+                    />
+                </Media.Figure>
+
+                <Media.Body className="bcs-AppActivity-body">
+                    {isMenuVisible && (
+                        <TetherComponent
+                            attachment="top right"
+                            className="bcs-AppActivity-confirm"
+                            constraints={[{ to: 'scrollParent', attachment: 'together' }]}
+                            targetAttachment="bottom right"
+                        >
+                            <Media.Menu isDisabled={isConfirmingDelete}>
+                                {canDelete && (
```

Comment:
Is this `canDelete` necessary if `isMenuVisible` already takes that into account?

Reply:
It's following the same convention as most of our other menus, but I can remove it.